src/common/FileIO.py

```diff
@@ -11,26 +11,76 @@
 Log = logger_instance
 
 
-def download_scores():
-    data = []
+# Logic for importing scores supporting any format changes
+def upgrade_score_export_format(data):
+    version = get_score_export_version(data)
+    # Upgrade file through all version upgrade functions
+    while version in score_version_upgrades:
+        data = score_version_upgrades[version](data)
+        if "version" not in data or data["version"] == version:
+            # Version did not increment, or some other error.
+            raise ValueError("Failed to upgrade score import format!")
+        version = data["version"]
+    return data
 
-    data.append(
-        {
-            "FileType": "leaders",
-            "contents": [DataStore.read_record("leaders", i) for i in range(DataStore.memory_map["leaders"]["count"])],
-        }
-    )
 
-    data.append(
-        {
-            "FileType": "tournament",
-            "contents": [DataStore.read_record("tournament", i) for i in range(DataStore.memory_map["tournament"]["count"])],
-        }
-    )
+def get_score_export_version(data):
+    if "version" in data:
+        return data["version"]
+    return 0
+
+
+# Version "0" was "classic score" format, before we versioned it.
+def score_export_0_to_1(data):
+    output = {"version": 1, "scores": {}}
+    # Convert to nicer format, with version number
+    for row in data:
+        output["scores"][row["FileType"]] = row["contents"]
+    return output
+
+
+# If we ever change the format again, we can use the below dictionary and functions named as so to update the format from older exports so they'll continue to work.
+# def score_export_1_to_2(data):
+
+current_score_version = 1
+score_version_upgrades = {0: score_export_0_to_1}
```

Comment:
Love the forward thinking. We try to do this sort of thing where we can....inevitably we need it where we don't put it.